File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/entity/Reaction.java
package at.ac.tuwien.sepm.groupphase.backend.entity;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
@Builder
@Entity

public class Reaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDateTime publishedAt;

    @Column(nullable = false)
    private ReactionType type;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "message_id", nullable = false)
    private Message message;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "owner_id", nullable = false)
    private ApplicationUser owner;

    public enum ReactionType {
        THUMBS_UP,
        THUMBS_DOWN
    }
}

File: frontend/src/app/dtos/reaction.ts
export enum ReactionType {
  THUMBS_UP = 'THUMBS_UP',
  THUMBS_DOWN = 'THUMBS_DOWN',
  NEUTRAL = 'NEUTRAL'
}

export class Reaction {
  constructor(
    public id: number,
    public messageId: number,
    public type: ReactionType,
    public owner: {id: number, name: string, email: string}) {
  }
}
